File: 1-intro/5-island-of-knowledge/cpp/boxBlur.cpp
vector<vector<int>> solution(vector<vector<int>> image) {
    int m = image.size() - 2;
    int n = image[0].size() - 2;

    vector<vector<int>> res(m, vector<int>(n));
    
    for (int i = 1; i < image.size() - 1; i++) {
        for (int j = 1; j < image[i].size() - 1; j++) {
            res[i - 1][j - 1] =
            (
                image[i - 1][j - 1] + image[i - 1][j] + image[i - 1][j + 1] +
                image[i][j - 1] + image[i][j] + image[i][j + 1] +
                image[i + 1][j - 1] + image[i + 1][j] + image[i + 1][j + 1]
            ) / 9;
        }        
    }

    return res;
}

File: 1-intro/5-island-of-knowledge/js/boxBlur.js
const solution = (img) => {
  let m = img.length - 2;
  let n = img[0].length - 2;
  
  let res = new Array(m);

  for (let i = 0; i < m; i++)
    res[i] = new Array(n);

  for (let i = 1; i < img.length - 1; i++) {
    for (let j = 1; j < img[i].length - 1; j++) {
        res[i - 1][j - 1] =
        Math.floor((
            img[i - 1][j - 1] + img[i - 1][j] + img[i - 1][j + 1] +
            img[i][j - 1] + img[i][j] + img[i][j + 1] +
            img[i + 1][j - 1] + img[i + 1][j] + img[i + 1][j + 1]
        ) / 9);
    }        
  }

  return res;
}
